modules/product/src/main/java/com/opengamma/strata/product/fxopt/FxDigitalOption.java

```diff
@@ -0,0 +1,641 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product.fxopt;
+
+import static com.opengamma.strata.collect.ArgChecker.inOrderOrEqual;
+
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.ImmutableValidator;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.basics.ReferenceData;
+import com.opengamma.strata.basics.Resolvable;
+import com.opengamma.strata.basics.currency.CurrencyPair;
+import com.opengamma.strata.product.Product;
+import com.opengamma.strata.product.common.LongShort;
+import com.opengamma.strata.product.fx.FxSingle;
+
+/**
+ * A vanilla Digital FX option.
+ * <p>
+ * An FX option is a financial instrument that provides an option based on the future value of
+ * a foreign exchange. The option is European, exercised only on the exercise date.
+ * <p>
+ * For example, a call on a 'EUR 1.00 / USD -1.41' exchange is the option to
+ * perform a foreign exchange on the expiry date, where USD 1.41 is paid to receive EUR 1.00.
+ */
+@BeanDefinition
```

Comment:
As it stands, all beans in the trade model should be generated as full beans, not light or minimal. If appropriate, the builder can be made private using builderStyle="private", but I don't think that applies here.